### ISSUE TITLE
Rubocop raises `Lint/UselessAssignment:` when calling `foo=(bar)` methods

### ISSUE THREAD

**Issue by glUk-skywalker** [bbatsov/rubocop#5883]
## Expected behavior

Rubocop doesn't raise `Lint/UselessAssignment:` when calling `foo=(bar)` methods

## Actual behavior

It does

## Steps to reproduce the problem

![image](https://user-images.githubusercontent.com/4256588/39971593-850ac6d0-5706-11e8-96c1-82cc3ee9c99f.png)

## RuboCop version
```
rubocop -V
0.55.0 (using Parser 2.5.1.0, running on ruby 2.5.0 x86_64-linux)
```

**Comment by glUk-skywalker** [comment]
invalid

**Comment by mikegee** [comment]
Just in case someone else sees this and wonders why it is invalid:

`keys = Buttons.accept_user(user)` defines a new local variable and does not call the `keys=` method. 

`self.keys = Buttons.accept_user(user)` is required here.
